JPS_VIRTUALSENSOR/AermodAgent/src/main/java/com/cmclinnovations/aermod/Aermod.java

```diff
@@ -413,6 +389,85 @@ public void createAERMODReceptorInput(Polygon scope, int nx, int ny, int simulat
+        public final double ylo;
+        public final double dy;
+
+        public AermodGrid(Polygon scope, int nx, int ny, int simulationSrid) {
+            List<Double> xDoubles = new ArrayList<>();
+            List<Double> yDoubles = new ArrayList<>();
+
+            for (int i = 0; i < scope.getCoordinates().length; i++) {
+
+                String originalSrid = "EPSG:4326";
+                double[] xyOriginal = { scope.getCoordinates()[i].x, scope.getCoordinates()[i].y };
+                double[] xyTransformed = CRSTransformer.transform(originalSrid, "EPSG:" + simulationSrid, xyOriginal);
+
+                xDoubles.add(xyTransformed[0]);
+                yDoubles.add(xyTransformed[1]);
+            }
+
+            this.xlo = Collections.min(xDoubles);
+            double xhi = Collections.max(xDoubles);
+            this.ylo = Collections.min(yDoubles);
+            double yhi = Collections.max(yDoubles);
+
+            this.dx = (xhi - this.xlo) / nx;
+            this.dy = (yhi - this.ylo) / ny;
+        }
+    }
+
+    public void createMockOutput(Polygon scope, int nx, int ny, int simulationSrid, PollutantType pollutantType,
+            int z) {
+
+        AermodGrid ag = new AermodGrid(scope, nx, ny, simulationSrid);
+        // Get the current date time
+        LocalDateTime currentTime = LocalDateTime.now();
+
+        // Format the current date
+        String formattedDate = currentTime.format(DateTimeFormatter.ofPattern("dd/MM/yy"));
+
+        // Format the current time
+        String formattedTime = currentTime.format(DateTimeFormatter.ofPattern("HH:mm:ss"));
+
+        StringBuilder sb = new StringBuilder(
+                "* AERMOD (22112 ): Emission dispersion simulation                                           ");
+        sb.append(formattedDate).append(System.lineSeparator());
+        sb.append("* AERMET ( 22112):                                                                          ")
+                .append(formattedTime).append(System.lineSeparator());
+        sb.append("* MODELING OPTIONS USED:   RegDFAULT  CONC  ELEV  FLGPOL  RURAL").append(System.lineSeparator());
+        sb.append("*         POST/PLOT FILE OF PERIOD VALUES FOR SOURCE GROUP: ALL     ")
+                .append(System.lineSeparator());
+        String logReceptor = String.format("*         FOR A TOTAL OF %d RECEPTORS.", nx * ny);
+        sb.append(logReceptor).append(System.lineSeparator());
+        sb.append("*         FORMAT: (2(1X,F13.5),1X,E13.6,3(1X,F8.2),2X,A6,2X,A8,2X,I8.8,2X,A8)")
+                .append(System.lineSeparator());
+        sb.append(
+                "*        X             Y      AVERAGE CONC    ZELEV    ZHILL    ZFLAG    AVE     GRP      NUM HRS   NET ID")
+                .append(System.lineSeparator());
+        sb.append(
+                "* ____________  ____________  ____________   ______   ______   ______  ______  ________  ________  ________")
+                .append(System.lineSeparator());
```

Comment:
I personally find the idea of creating mock outputs from AERMOD unnecessary and misleading, if there are no emission sources and AERMOD cannot run, then the simulation should be skipped entirely or provide null values (not giving a bunch of zeroes, i.e. giving the impression of perfectly clean air).

If this is absolutely necessary I would rather create the mock raster file output directly, rather than trying to mimic the outputs from AERMOD. It is also unnecessary to write the first lines exactly as they appear, you just have to create a number of blank lines, then the header, then write the values, comma separated might even work.